infra/default.nix

```diff
@@ -0,0 +1,10 @@
+# Running `nix-shell` on this file puts you in an environment where all the
+# dependencies needed to work on this package using `Cabal` are available.
+#
+# Running `nix-build` builds this package.
+let
+  drv = (import ../. {}).cardano-sl-infra;
+in
+  if ((import <nixpkgs> {}).stdenv.lib.inNixShell)
```

Comment:
`(import ../. {}).pkgs.lib.inNixShell` is more pure

Reply:
Fixed